rest/utilities_testing_resttester.go

```diff
@@ -491,3 +496,45 @@ func (rt *RestTester) PutDocWithAttachment(docID string, body string, attachment
 	}
 	return rt.PutDocDirectly(docID, rawBody)
 }
+
+type RawDocResponse struct {
+	Xattrs RawDocXattrs `json:"_xattrs"`
+}
+
+// RawDocXattrs is a non-exhaustive set of xattrs returned by the _raw endpoint. Used for test assertions.
+type RawDocXattrs struct {
+	RawDocXattrsWellKnown
+	RawDocXattrsOthers
+}
+
+// RawDocXattrsWellKnown contains well known fields - see RawDocXattrsOthers for those not known until runtime.
+type RawDocXattrsWellKnown struct {
+	Sync       db.SyncDataJSON `json:"_sync"`
+	GlobalSync map[string]any  `json:"_globalSync"`
+	VV         map[string]any  `json:"_vv"`
+	MOU        map[string]any  `json:"_mou"`
+}
+
+// RawDocXattrsOthers contains fields that can only be known at runtime (e.g. user xattr names)
+type RawDocXattrsOthers map[string]interface{}
+
+func (t *RawDocXattrs) MarshalJSON() ([]byte, error) {
+	// overwrite fields in Main into Extra
+	data, err := json.Marshal(t.RawDocXattrsWellKnown)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(data, &t.RawDocXattrsOthers); err != nil {
```

Comment:
This seems very weird to me that `RawDocXattrs` has side effects. It doesn't feel right for caching, especially in a test function, and you don't know someone has mutated `RawDocXattrs`. Are you trying to avoid a copy of `t.RawDocXattrsOthers`?

Reply:
Dumb implementation of JSON v2's `inline` feature